core/src/main/java/org/geysermc/geyser/GeyserLogger.java

```diff
@@ -103,6 +113,14 @@ default void debug(@Nullable Object object) {
         debug(String.valueOf(object));
     }
 
+    /**
+     * Logs a message with parameters if debug mode is enabled.
+     *
+     * @param message the message with placeholders for the objects
+     * @param parameters the parameters for the message
+     */
+    void debug(String message, Object... parameters);
```

Comment:
Can we just make this a `default` and check `isDebug` instead of the debug variable?

Reply:
Added - i've left in overrides for the method for loggers that have a similar method built-in.